mast/tools/sis_authentication.py

```diff
@@ -0,0 +1,118 @@
+import requests
+from bs4 import BeautifulSoup
+import re
+
+
+def authenticate_via_sis(name, surname, login=None, ukco=None, is_employee=False):
+    """
+    Simple authentication via sis.
+    Employee must submit: name, surname
+    Student must submit: name, surname, (login or ukco)
+    :param name: name of the person - obligatory:
+    :param surname: surname of the person - obligatory:
+    :param login: login for students
+    :param ukco: UKCO for students
+    :param is_employee: bool if the person is employee
+    :return: True if such person exists, False otherwise
+    """
+    if is_employee:
+        if name == None or surname == None:
+            # expecting both name and surname
+            print('Expected both name and surname\nName={}\nSurname={}'.format(name, surname))
+            return False
+        return authenticate_employee(name=name, surname=surname)
+    else:
+        if name == None or surname == None:
```

Comment:
DRY principle :). It would suite to put the following lines into a method.

Reply:
I like it better here. It shows what error it is right away and explicitly shows that the function ends if its wrong.

Reply:
Good argument, you defended your approach ;)